app/src/main/assets/GimelGimel/CesiumEventHandler.js

```diff
@@ -48,43 +49,60 @@ GG.EventHandler.prototype.setScreenSpaceEventAction = function (screenSpaceEvent
 };
 
 
-GG.EventHandler.prototype.setSingleTouchActions = function (layersManager) {
-
-    var handler = this;
-    //Set mouse move event listener
-    //Cesium.ScreenSpaceEventType.LEFT_DOWN is mapped to touch events.
-    this.setScreenSpaceEventAction(Cesium.ScreenSpaceEventType.LEFT_DOWN, function (movement) {
-        var entity = handler.pickEntity(movement.position);
-        clickEntityWithLayer(entity);
-
-        var cartesian = handler._viewer.camera.pickEllipsoid(movement.position, handler._scene.globe.ellipsoid);
-        if (cartesian) {
-            var cartographic = Cesium.Cartographic.fromCartesian(cartesian);
-            var longitude = Cesium.Math.toDegrees(cartographic.longitude);
-            var latitude = Cesium.Math.toDegrees(cartographic.latitude);
-
-            GG.AndroidAPI.updateSelectedLocation({
-                            longitude: longitude,
-                            latitude: latitude
-                        });
-
-
-        }
-    });
+GG.EventHandler.prototype.onSingleTap = function (relativeX, relativeY) {
+    var self = this;
+    var position = self.getWindowPosition(relativeX, relativeY);
+    var entity = this.pickEntity(position);
+    clickEntityWithLayer(entity);
 
     function clickEntityWithLayer(entity) {
-        $.each(layersManager.getLayers(), function (layerId, layer) {
+        $.each(GG.layerManager.getLayers(), function (layerId, layer) {
             try {
                 if (layer.getEntity(entity.id)) {
                     GG.AndroidAPI.onEntityClicked(layerId, entity.id)
                 }
             } catch (err) {
-//                    iterate to next layer
+//              iterate to next layer
             }
         });
     }
 };
 
+GG.EventHandler.prototype.onLongPress = function (relativeX, relativeY) {
+    var self = this;
```

Comment:
There is no need to declare this variable, you should do it if you need "this" inside inner / async method.